cli/azd/cmd/container.go

```diff
@@ -523,11 +528,53 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 	container.MustRegisterSingleton(config.NewUserConfigManager)
 	container.MustRegisterSingleton(config.NewManager)
 	container.MustRegisterSingleton(config.NewFileConfigManager)
-	container.MustRegisterScoped(func() auth.ExternalAuthConfiguration {
-		return auth.ExternalAuthConfiguration{
-			Endpoint: os.Getenv("AZD_AUTH_ENDPOINT"),
-			Key:      os.Getenv("AZD_AUTH_KEY"),
+	container.MustRegisterScoped(func() (auth.ExternalAuthConfiguration, error) {
+		cert := os.Getenv("AZD_AUTH_CERT")
+		endpoint := os.Getenv("AZD_AUTH_ENDPOINT")
+		key := os.Getenv("AZD_AUTH_KEY")
+
+		client := &http.Client{}
+		if len(cert) > 0 {
+			certBytes, decodeErr := base64.StdEncoding.DecodeString(cert)
+			if decodeErr != nil {
+				return auth.ExternalAuthConfiguration{},
+					fmt.Errorf("failed to decode provided AZD_AUTH_CERT: %w", decodeErr)
+			}
+
+			cert, certParseErr := x509.ParseCertificate(certBytes)
+			if certParseErr != nil {
+				return auth.ExternalAuthConfiguration{},
+					fmt.Errorf("failed to decode provided AZD_AUTH_CERT: %w", certParseErr)
+			}
+
+			caCertPool := x509.NewCertPool()
+			caCertPool.AddCert(cert)
+			tlsConfig := &tls.Config{
+				RootCAs:    caCertPool,
+				MinVersion: tls.VersionTLS12,
+			}
+
+			transport := http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
Just wondering - why this vs just `&http.Transport{TLSClientConfig: tlsConfig}`.  I assume that the DefaultTransport has other state we want to inherit/share but I wonder what that is.

Reply:
The connection timeouts and number of idled connections are all useful and more sane defaults than when not provided. [k8s](https://github.com/kubernetes/kubernetes/blob/b722d017a34b300a2284b890448e5a605f21d01e/staging/src/k8s.io/apimachinery/pkg/util/net/http.go#L105C5-L105C22) does something similar when overriding TLS client configs.